storage/src/vespa/storage/persistence/apply_bucket_diff_entry_complete.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "apply_bucket_diff_entry_complete.h"
+#include <cassert>
+
+namespace storage {
+
+ApplyBucketDiffEntryComplete::ApplyBucketDiffEntryComplete(ResultPromise result_promise, const framework::Clock& clock, metrics::DoubleAverageMetric& latency_metric)
+    : _result_handler(nullptr),
+      _result_promise(std::move(result_promise)),
+      _start_time(clock),
+      _latency_metric(latency_metric)
+{
+}
+
+ApplyBucketDiffEntryComplete::~ApplyBucketDiffEntryComplete() = default;
+
+void
+ApplyBucketDiffEntryComplete::onComplete(std::unique_ptr<spi::Result> result)
+{
+    if (_result_handler != nullptr) {
+        _result_handler->handle(*result);
+    }
+    _result_promise.set_value(std::move(result));
+    _latency_metric.addValue(_start_time.getElapsedTimeAsDouble());
```

Comment:
Can `onComplete` for different document instances in the same merge be called from multiple threads concurrently? If so, this might race. Consider passing the elapsed time to the result future and aggregate this as part of the async result processing in the main merge handling thread.